matroid/src/collections.py

```diff
@@ -0,0 +1,176 @@
+            'name': name,
+            'url': url,
+            'sourceType': source_type
+        }
+        return requests.request(method, endpoint, **{'headers': headers, 'data': data})
+    except Exception as e:
+        raise error.APIConnectionError(message=e)
+
+
+@api_call(error.InvalidQueryError)
+def delete_collection_task(self, task_id):
+    """Deletes a completed collection mananger task"""
+    (endpoint, method) = self.endpoints['delete_collection_task']
+    endpoint = endpoint.replace(':key', task_id)
+
+    try:
+        headers = {'Authorization': self.token.authorization_header()}
+        return requests.request(method, endpoint, **{'headers': headers})
+    except Exception as e:
+        raise error.APIConnectionError(message=e)
+
+
+@api_call(error.InvalidQueryError)
+def delete_collection(self, collection_id):
+    """Deletes a collection with no active indexing tasks"""
+    (endpoint, method) = self.endpoints['delete_collection']
+    endpoint = endpoint.replace(':key', collection_id)
+
+    try:
+        headers = {'Authorization': self.token.authorization_header()}
+        return requests.request(method, endpoint, **{'headers': headers})
+    except Exception as e:
+        raise error.APIConnectionError(message=e)
+
+
+@api_call(error.InvalidQueryError)
+def get_collection_task(self, task_id):
+    """Creates a new collection from a web or S3 url"""
+    (endpoint, method) = self.endpoints['get_collection_task']
+    endpoint = endpoint.replace(':key', task_id)
+
+    try:
+        headers = {'Authorization': self.token.authorization_header()}
+        return requests.request(method, endpoint, **{'headers': headers})
+    except Exception as e:
+        raise error.APIConnectionError(message=e)
+
+
+@api_call(error.InvalidQueryError)
+def get_collection(self, collection_id):
+    """Get information about a specific collection"""
+    (endpoint, method) = self.endpoints['get_collection']
+    endpoint = endpoint.replace(':key', collection_id)
+
+    try:
+        headers = {'Authorization': self.token.authorization_header()}
+        return requests.request(method, endpoint, **{'headers': headers})
+    except Exception as e:
+        raise error.APIConnectionError(message=e)
+
+
+@api_call(error.InvalidQueryError)
+def kill_collection_index(self, task_id, include_collection_info):
+    """Kills an active collection indexing task"""
+    (endpoint, method) = self.endpoints['kill_collection_index']
+    endpoint = endpoint.replace(':key', task_id)
+
+    try:
+        headers = {'Authorization': self.token.authorization_header()}
+        data = {
+            'includeCollectionInfo': 'true' if include_collection_info else ''
+        }
+        return requests.request(method, endpoint, **{'headers': headers, 'data': data})
+    except Exception as e:
+        raise error.APIConnectionError(message=e)
+
+
+@api_call(error.InvalidQueryError)
+def query_by_detection_scores(self, task_id, thresholds, num_results):
```

Comment:
Let's add some top level comments for these functions as well. If it isn't clear, `query_by_detection_scores` takes in a map of thresholds, and returns media in the collection with detections above those thresholds. `query_by_image` takes in an image file or url and returns similar media from the collection..

Reply:
Added!